boulder-va: panic during deploy of release-2025-05-12

I updated the boulder-va first before moving onto the RVA software and received this from one of the boulder-va processes.

```
goroutine 483 [running]:
github.com/letsencrypt/boulder/va.(*ValidationAuthorityImpl).DoCAA(0xc0001a2b00, {0x1bc3520, 0xc00026e700}, 0xc00026e690)
	/tmp/tmps3sif24m/src/github.com/letsencrypt/boulder/va/caa.go:111 +0x563
github.com/letsencrypt/boulder/va/proto._CAA_DoCAA_Handler.func1({0x1bc3520?, 0xc00026e700?}, {0x1849440?, 0xc00026e690?})
	/tmp/tmps3sif24m/src/github.com/letsencrypt/boulder/va/proto/va_grpc.pb.go:204 +0xcb
github.com/letsencrypt/boulder/grpc.(*serverMetadataInterceptor).Unary(0xc000651350, {0x1bc3478, 0xc000f08540}, {0x1849440, 0xc00026e690}, 0x0?, 0xc000e03620)
	/tmp/tmps3sif24m/src/github.com/letsencrypt/boulder/grpc/interceptors.go:124 +0x29d
google.golang.org/grpc.getChainUnaryHandler.func1({0x1bc3478, 0xc000f08540}, {0x1849440, 0xc00026e690})
	/tmp/tmps3sif24m/src/github.com/letsencrypt/boulder/vendor/google.golang.org/grpc/server.go:1217 +0xa8
github.com/letsencrypt/boulder/grpc.(*authInterceptor).Unary(0x38?, {0x1bc3478, 0xc000f08540}, {0x1849440, 0xc00026e690}, 0xc00072aba0?, 0xc000f0a140)
	/tmp/tmps3sif24m/src/github.com/letsencrypt/boulder/grpc/interceptors.go:492 +0x67
google.golang.org/grpc.getChainUnaryHandler.func1({0x1bc3478, 0xc000f08540}, {0x1849440, 0xc00026e690})
	/tmp/tmps3sif24m/src/github.com/letsencrypt/boulder/vendor/google.golang.org/grpc/server.go:1217 +0xa8
github.com/letsencrypt/boulder/grpc.(*serverBuilder).Build.(*ServerMetrics).UnaryServerInterceptor.UnaryServerInterceptor.func8({0x1bc3478, 0xc000f08540}, {0x1849440, 0xc00026e690}, 0xc000e117a0?, 0xc000f0a100)
	/tmp/tmps3sif24m/src/github.com/letsencrypt/boulder/vendor/github.com/grpc-ecosystem/go-grpc-middleware/v2/interceptors/server.go:22 +0x275
google.golang.org/grpc.NewServer.chainUnaryServerInterceptors.chainUnaryInterceptors.func1({0x1bc3478, 0xc000f08540}, {0x1849440, 0xc00026e690}, 0xc000e117a0, 0x80?)
	/tmp/tmps3sif24m/src/github.com/letsencrypt/boulder/vendor/google.golang.org/grpc/server.go:1208 +0x7c
github.com/letsencrypt/boulder/va/proto._CAA_DoCAA_Handler({0x18b3420, 0xc0001a2b00}, {0x1bc3478, 0xc000f08540}, 0xc000e20d00, 0xc000668400)
	/tmp/tmps3sif24m/src/github.com/letsencrypt/boulder/va/proto/va_grpc.pb.go:206 +0x143
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000654400, {0x1bc3478, 0xc000f08420}, 0xc00097f140, 0xc0006515f0, 0x26c79f0, 0x0)
	/tmp/tmps3sif24m/src/github.com/letsencrypt/boulder/vendor/google.golang.org/grpc/server.go:1405 +0x1036
google.golang.org/grpc.(*Server).handleStream(0xc000654400, {0x1bc3da8, 0xc0007cc340}, 0xc00097f140)
	/tmp/tmps3sif24m/src/github.com/letsencrypt/boulder/vendor/google.golang.org/grpc/server.go:1815 +0xb88
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	/tmp/tmps3sif24m/src/github.com/letsencrypt/boulder/vendor/google.golang.org/grpc/server.go:1035 +0x7f
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 294
	/tmp/tmps3sif24m/src/github.com/letsencrypt/boulder/vendor/google.golang.org/grpc/server.go:1046 +0x11d
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x10831c3]
```

--- a/va/caa.go
+++ b/va/caa.go
@@ -108,7 +108,7 @@ func (va *ValidationAuthorityImpl) DoCAA(ctx context.Context, req *vapb.IsCAAVal
 	if internalErr != nil {
 		logEvent.InternalError = internalErr.Error()
 		prob = detailedError(internalErr)
-		prob.Detail = fmt.Sprintf("While processing CAA for %s: %s", req.Identifier.Value, prob.Detail)
+		prob.Detail = fmt.Sprintf("While processing CAA for %s: %s", ident.Value, prob.Detail)
 	}
 
 	if va.isPrimaryVA() {
@@ -125,7 +125,7 @@ func (va *ValidationAuthorityImpl) DoCAA(ctx context.Context, req *vapb.IsCAAVal
 		if remoteProb != nil {
 			prob = remoteProb
 			va.log.Infof("CAA check failed due to remote failures: identifier=%v err=%s",
-				req.Identifier.Value, remoteProb)
+				ident.Value, remoteProb)
 		}
 	}
 

